include/getfilecontents.h

```diff
@@ -21,11 +22,23 @@
   SOFTWARE.
 */
 
+/**
+ * @file getfilecontents.h
+ * @brief Fills the econf_file struct with data from a given file.
+ */
+
 #pragma once
 
 #include "libeconf.h"
 #include "keyfile.h"
 
-/* Fill the econf_file struct with values from the given file */
+/**
+ * @brief Read the file line by line and parse for comments, keys and values.
+ * @param read_file The econf_file struct where the data will be stored.
+ * @param file The file to read from.
+ * @param delim A char used to assign a value to a key.
```

Comment:
That doesn't sound right :-)
This is basically the key / value delimiter, I definitely prefer the description from `econf_readFile`. It should also be mentioned what this separator means - will whitespace before / after this character be ignored? Can I give multiple characters?